### PR TITLE
Backend Event Participation

### PR DESCRIPTION
**Pull request by fileames**
Endpoints related to event participation are implemented.
These include:
- sending request to participate an event
- attending as spectator
- accepting and rejecting interested users
- deleting any participating status(interested, spectator, participating)
- getting event interesteds, participants, spectators

Activity stream response is returned on accepting and rejecting participants.
Get requests return the SportsEvent schema with related properties.

### CODE REVIEW

File: back-end/sports_platform/sports_platform_api/models/event_models.py [hunk 1]
```diff
@@ -61,31 +66,333 @@ def create_event(data):
...
 class EventParticipationRequesters(models.Model):
     class Meta:
         db_table = 'event_participation_requesters'
         unique_together = (('user', 'event'),)
 
-    user = models.ForeignKey('User', on_delete=models.CASCADE)
-    event = models.ForeignKey('Event', on_delete=models.CASCADE)
+    user = models.ForeignKey('User', related_name='interested_events', on_delete=models.CASCADE)
+    event = models.ForeignKey('Event', related_name='interested_users', on_delete=models.CASCADE)
+    message = models.TextField()
```
Comment (berkatil) [inline back-end/sports_platform/sports_platform_api/models/event_models.py:397]: I think this can be blank

**Review by berkatil — CHANGES_REQUESTED** [review]
Thanks a lot for your great effort!

File: back-end/sports_platform/sports_platform_api/models/event_models.py [hunk 2]
```diff
@@ -61,31 +66,333 @@ def create_event(data):
...
+        data_dict['items'] = []
+
+        utc_dt = datetime.now(timezone.utc)  # UTC time
+        dt = utc_dt.astimezone()
+        try:
+            with transaction.atomic():
+                for user in accept_user_id_list:
+                    if num_remaining_places <= 0:
+                        data_dict['total_items'] = len(data_dict['items'])
+                        return data_dict
```
[the remark this answers is not in the document]
Comment (fileames) [reply back-end/sports_platform/sports_platform_api/models/event_models.py]: Good point, fixed it.

File: back-end/sports_platform/sports_platform_api/models/event_models.py (hunk 1 above)
Comment (fileames) [reply back-end/sports_platform/sports_platform_api/models/event_models.py]: Added the blank parameter.

Comment (berkatil) [reply back-end/sports_platform/sports_platform_api/models/event_models.py]: thanks

**Comment by berkatil** [comment]
Thanks a lot it is ready to be merged

**Pull request merged**